CMakeModules/AFconfigure_forge_submodule.cmake

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2019, ArrayFire
+# All rights reserved.
+#
+# This file is distributed under 3-clause BSD license.
+# The complete license agreement can be obtained at:
+# http://arrayfire.com/licenses/BSD-3-Clause
+
+set(ArrayFireInstallPrefix ${CMAKE_INSTALL_PREFIX})
+set(ArrayFireBuildType ${CMAKE_BUILD_TYPE})
+set(CMAKE_INSTALL_PREFIX ${ArrayFire_BINARY_DIR}/extern/forge/package)
+set(CMAKE_BUILD_TYPE Release)
+set(FG_BUILD_EXAMPLES OFF CACHE BOOL "Used to build Forge examples")
+set(FG_BUILD_DOCS OFF CACHE BOOL "Used to build Forge documentation")
+set(FG_WITH_FREEIMAGE OFF CACHE BOOL "Turn on usage of freeimage dependency")
+add_subdirectory(extern/forge EXCLUDE_FROM_ALL)
```

Comment:
`add_subdirectory` should stay in the top level CMake. Also it would be nice if this was in a function

Reply:
Is this any different ? because `include` basically just replaces the content into here right ?

Reply:
Yes but it kind of hides the details. It would be better to have all the subdirectories in one location